src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala

```diff
@@ -5,16 +5,41 @@ import java.util.UUID
 import io.circe.generic.auto._
 import sangria.macros.derive._
 import sangria.marshalling.circe._
-import sangria.schema.{Argument, Field, InputObjectType, LongType, ObjectType, OptionType, fields}
+import sangria.schema.{Argument, Field, InputObjectType, IntType, ObjectType, OptionType, fields}
 import uk.gov.nationalarchives.tdr.api.auth.{ValidateSeries, ValidateUserOwnsConsignment}
-import uk.gov.nationalarchives.tdr.api.graphql.ConsignmentApiContext
 import uk.gov.nationalarchives.tdr.api.graphql.fields.FieldTypes._
+import uk.gov.nationalarchives.tdr.api.graphql.{ConsignmentApiContext, DeferFileProgress, DeferTotalFiles}
 
 object ConsignmentFields {
   case class Consignment(consignmentid: Option[UUID] = None, userid: UUID, seriesid: UUID)
   case class AddConsignmentInput(seriesid: UUID)
+  case class AntivirusProgress(processedFiles: Int)
+  case class FileCheckProgress(antivirusProgress: AntivirusProgress)
+
+  implicit val FileCheckProgressType: ObjectType[Unit, FileCheckProgress] =
+    deriveObjectType[Unit, FileCheckProgress]()
+  implicit val AntivirusProgressType: ObjectType[Unit, AntivirusProgress] =
+    deriveObjectType[Unit, AntivirusProgress]()
+
+  implicit private val ConsignmentType: ObjectType[Unit, Consignment] = ObjectType(
+    "Consignment",
+    fields[Unit, Consignment](
+      Field("consignmentid", OptionType(UuidType), resolve = _.value.consignmentid),
+      Field("userid", UuidType, resolve = _.value.userid),
+      Field("seriesid", UuidType, resolve = _.value.seriesid),
+      Field(
+        "totalFiles",
+        IntType,
+        resolve = context => DeferTotalFiles(context.value.consignmentid)
+      ),
+      Field(
+        "fileCheckProgress",
```

Comment:
The nested structure has made me think we should actually put more of the file check results into this object. So something like:

```
fileChecks {
  antivirus {
    filesProcessed
  },
  fileFormatId {
    filesProcessed
  }
}
```

Which would let us add the results into each section as well as the number of files processed. What do you think, and how much work would it be to shuffle the result classes around to fit that pattern?

Reply:
I think that's a good way forward, I was picturing something similar when I started the nested stuff. I don't think it would be too difficult to add in the other processes once we're at that point. I've made the classes a bit more general rather than all about AV and the AV object rests inside of the fileChecks object so I think all we need to do is change that case class to include the objects, not many other big changes I can think of though!